libraries/AP_Terrain/AP_Terrain.cpp

```diff
@@ -215,16 +215,11 @@ bool AP_Terrain::height_above_terrain(float &terrain_altitude, bool extrapolate)
     if (!height_terrain_difference_home(terrain_difference, extrapolate)) {
         return false;
     }
-    Location loc;
-    if (!ahrs.get_position(loc)) {
```

Comment:
Yeah I can see why checking for a location first is still worthwhile.

I'm not sure where you get the idea that we are returning altitude above EKF origin if we aren't getting a home location? I agree that if home.alt == 0 then yes, we are returning the height above the EKF origin, but thats at least more valid then the old behavior that I changed which was to return your relative home altitude as absolute altitude. (And again I believe every caller locks a home before EKF can lock an origin, so it should still be fine)